### PR TITLE
chore: optimize pipeline detail api

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

1. stack details add **outputs** and delete **url**
2. pipeline extended information(dns, endpoint, dashboard etc.) read from stack details, reduce `CFN.describeStack `requests 

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy control plane with CloudFront + S3 + API gateway
  - [ ] deploy control plane within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

### CODE REVIEW

File: src/control-plane/backend/lambda/api/common/utils.ts [hunk 1]
```diff
@@ -367,6 +367,44 @@ function getValueFromStackOutputSuffix(pipeline: IPipeline, stackType: PipelineS
   return `#.${stackName}.${suffix}`;
 }
 
+function getInfoFromPipelineStatus(status: PipelineStatus, stackType: PipelineStackType, key: string): string {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/common/utils.ts:370]: rename it to `getStackOutputFromPipelineStatus` ?

File: src/control-plane/backend/lambda/api/common/types.ts [hunk 2]
```diff
@@ -270,7 +270,7 @@ export interface PipelineStatusDetail {
   readonly stackType: PipelineStackType;
   readonly stackStatus: StackStatus | undefined;
   readonly stackStatusReason: string;
-  readonly url: string;
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/common/types.ts:273]: Is it used by frontend?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/common/types.ts (hunk 2 above)
Comment (tyyzqmf) [reply src/control-plane/backend/lambda/api/common/types.ts:273]: No, frontend build all urls by self .

File: src/control-plane/backend/lambda/api/common/utils.ts (hunk 1 above)
Comment (tyyzqmf) [reply src/control-plane/backend/lambda/api/common/utils.ts]: done

**Review by zxkane — APPROVED** [review]

**Pull request merged**